### PR TITLE
fix(TFD-2647): Add type converter.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**What is the current behavior?** (You can also link to an open issue here)

The type converter is deactivated.

**What is the new behavior?**

The type converter is activated.

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Other information**:

The two PR are related:
https://github.com/Talend/daikon/pull/342
https://github.com/Talend/components/pull/1158

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1701/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1701/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1702/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1702/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1728/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1728/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1730/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1730/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1731/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1731/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1735/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1735/)

File: components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/aggregate/AggregateCombineFn.java [hunk 1]
```diff
@@ -432,8 +432,8 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
             if (avroSchemaStr != null) {
                 valueCoder = AvroCoder.of(new Schema.Parser().parse(avroSchemaStr));
             }
-            return avroSchemaStr == null ? ListCoder.of(NullableCoder.of(StringUtf8Coder.of()))
-                    : ListCoder.of(NullableCoder.of(valueCoder));
+            return (Coder<List>) (avroSchemaStr == null ? ListCoder.of(NullableCoder.of(StringUtf8Coder.of()))
+                    : ListCoder.of(NullableCoder.of(valueCoder)));
```
Comment (RyanSkraba) [inline components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/aggregate/AggregateCombineFn.java]: Quick note -- the original code compiles correctly for Java 7 but causes an error for Java 8.  This is unrelated to the conversion story, except that it was simpler to fix here.  The conversion story requires Java 8 API objects (LocalTime, etc).

File: components/components-processing/processing-runtime/src/test/java/org/talend/components/processing/runtime/typeconverter/TypeConverterUtilsTest.java [hunk 2]
```diff
@@ -263,106 +264,146 @@ public void testConvertSchema() {
...
+    @Test
+    public void testConvertValueToDoubleWithFormat2() {
+        testConvertValue(Schema.create(Schema.Type.STRING), "123,456.78", TypeConverterProperties.TypeConverterOutputTypes.Double,
+                "#,####.##", Double.class);
     }
 
     @Test
     public void testConvertValueToDecimalNoFormat() {
-        testConvertValue("3.5", TypeConverterProperties.TypeConverterOutputTypes.Decimal, null, BigDecimal.class);
+        // testConvertValue("3.5", TypeConverterProperties.TypeConverterOutputTypes.Decimal, null, BigDecimal.class);
```
Comment (pbailly) [inline components/components-processing/processing-runtime/src/test/java/org/talend/components/processing/runtime/typeconverter/TypeConverterUtilsTest.java]: Please ignore or remove the test

File: components/components-processing/processing-runtime/src/test/java/org/talend/components/processing/runtime/typeconverter/TypeConverterUtilsTest.java [hunk 3]
```diff
@@ -263,106 +264,146 @@ public void testConvertSchema() {
...
     }
 
     @Test
     public void testConvertValueToDecimalWithFormat() {
-        testConvertValue("1,234", TypeConverterProperties.TypeConverterOutputTypes.Decimal, "#,###", BigDecimal.class);
+        // testConvertValue("1,234", TypeConverterProperties.TypeConverterOutputTypes.Decimal, "#,###", BigDecimal.class);
```
Comment (pbailly) [inline components/components-processing/processing-runtime/src/test/java/org/talend/components/processing/runtime/typeconverter/TypeConverterUtilsTest.java]: Please ignore or remove the test

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1741/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1741/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1742/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1742/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1757/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1757/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1758/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1758/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1767/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1767/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1768/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1768/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1773/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1773/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

File: components/components-processing/processing-definition/src/main/java/org/talend/components/processing/definition/ProcessingFamilyDefinition.java [hunk 4]
```diff
@@ -46,11 +46,8 @@
     public ProcessingFamilyDefinition() {
         super(NAME,
                 // Components
-                new FilterRowDefinition(), new PythonRowDefinition(), new ReplicateDefinition(),
-                new TypeConverterDefinition(), new FieldSelectorDefinition(),
-                new WindowDefinition(), new NormalizeDefinition(), new AggregateDefinition()
-        // Component wizards
-        );
+                new AggregateDefinition(), new FieldSelectorDefinition(), new FilterRowDefinition(), new NormalizeDefinition(),
+                new PythonRowDefinition(), new ReplicateDefinition(), new TypeConverterDefinition(), new WindowDefinition());
```
Comment (pbailly) [inline components/components-processing/processing-definition/src/main/java/org/talend/components/processing/definition/ProcessingFamilyDefinition.java]: Please, be careful, you may have conflict with https://github.com/Talend/components/pull/1174 .

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1778/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1778/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1781/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1781/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1782/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1782/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1783/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1783/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1788/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1788/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-2647-typeconverter)
